enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,180 @@
+To perform the PromQL request, the update service will be extended with configuration for the Prometheus connection, including a URI and authentication credentials.
+
+To reliably and efficiently return cluster-specific responses, the ID set returned by configured queries may be cached.
+There are a few possibilities for caching; although selecting a particular caching implementation is up to the update service's maintainers.
+The policy engine should refresh the cache on cache misses, as discussed in [the schema section](#enhanced-graph-data-schema-for-blocking-edges).
+
+For every graph-builder response, the policy engine may aggregate queries for all edges and warm the cache of new queries.
+This will make the initial client request faster, at the expense of maintaining a cache that may not be needed before it goes stale.
+
+The policy engine should also be prepared for, and alert on, misconfigured PromQL that results in request failures.
+In the event of such a failure, the fallback behavior should be blocking the relevant edge for all clusters until the misconfiguration is fixed.
+
+### User Stories
+
+#### Bugs which impact a subset of clusters
+
+As described in [the *Motivation* section](#motivation), enabling things like "we'd like to block this edge for clusters born in 4.1 with the `None` platform".
+
+#### OpenShift Dedicated
+
+[OpenShift Dedicated][dedicated] (OSD) restricts the recommended update sets for managed clusters more aggressively than the default service.
+For example, OSD blocked edges into 4.3.19 temporarily while [rhbz#1838007][rhbz-1838007] was investigated, while the default service did not.
+With this enhancement, OSD-specific edge blocking could be accomplished with entries like:
+
+```yaml
+to: 4.3.19
+from: .*
+clusters:
+  # show unmanaged clusters the edge, but block it from managed clusters
+  prometheus: |
+    max by (_id) (0*subscription_labels{managed="true"})
+    or
+    max by (_id) (subscription_labels{managed="false"})
+```
+
+The query returns `{_id="..."}=0` results for `managed="true"` clusters (blocking the edge) and `{_id="..."}=1` results for `managed="false"` clusters (allowing the edge).
+Clusters with both `managed="true"` and `managed="false"` records in the current time window will fall into the safer "block" bucket (the `or` semantics are defined [here][PromQL-or]).
+
+### Risks and Mitigations
+
+There is a risk that [published graph-data information](distribute-secondary-metadata-as-container-image.md) which assumes access to Red Hat's internal Telemetry will block edges when consumed by update services without access to the internal Telemetry.
+The risk is small initially, when we have no such edges.
+If targeted blocking becomes widespread, a simple but tedious mitigation would be asking users to maintain their own graph-data (e.g. removing all targeted edge-blocking references) and build their own graph-data image.
```

Comment:
I do not think this is an option. We need turn key solutions for our customers and I would be against customers modifying the yamls themselves.

Reply:
The alternative is "set up your own local Prometheus aggregator", which sounds like a reasonable long-term approach to me.  Do you see additional alternatives that we should cover here?  Are you comfortable with the local-Prom alternative?

Reply:
Ah, "disable the targeted-edge-blocking plugin" is another path.  Updated to discuss this with 2037152 -> b78cced.

Reply:
I could see users with a mix of local cluster flavors wanting to deploy a local aggregation plugin, or wanting to block all conditional edges and manually workaround any dead-ends that introduced.  I don't think this enhancement needs to have opinions about which approaches folks running update services should take; it just lays out the options and explains the trade-offs.